diff --git a/python_orocos_kdl/cmake/FindSIP.cmake b/python_orocos_kdl/cmake/FindSIP.cmake
index 53e2888..25b70b2 100644
--- a/python_orocos_kdl/cmake/FindSIP.cmake
+++ b/python_orocos_kdl/cmake/FindSIP.cmake
@@ -31,7 +31,7 @@ IF(SIP_VERSION)
   SET(SIP_FOUND TRUE)
 ELSE(SIP_VERSION)
 
-  FIND_FILE(_find_sip_py FindSIP.py PATHS ${CMAKE_MODULE_PATH})
+  FIND_FILE(_find_sip_py FindSIP.py PATHS ${CMAKE_MODULE_PATH} NO_CMAKE_FIND_ROOT_PATH)
 
   EXECUTE_PROCESS(COMMAND ${PYTHON_EXECUTABLE} ${_find_sip_py} OUTPUT_VARIABLE sip_config)
   IF(sip_config)
